src/wh_server_crypto.c

```diff
@@ -329,18 +352,18 @@ static int hsmCacheKeyEcc(whServerContext* server, ecc_key* key, whKeyId* outId)
 static int hsmLoadKeyEcc(whServerContext* server, ecc_key* key, uint16_t keyId,
     int curveId)
 {
+    uint8_t* cacheBuf;
+    whNvmMetadata* cacheMeta;
     int ret;
-    int slotIdx = 0;
     uint32_t keySz;
     keyId |= WOLFHSM_KEYTYPE_CRYPTO;
     /* freshen the key */
-    ret = slotIdx = hsmFreshenKey(server, keyId);
+    ret = hsmFreshenKey(server, keyId, &cacheBuf, &cacheMeta);
     /* decode the key */
-    if (ret >= 0) {
-        keySz = server->cache[slotIdx].meta->len / 3;
-        ret = wc_ecc_import_unsigned(key, server->cache[slotIdx].buffer,
-            server->cache[slotIdx].buffer + keySz,
-            server->cache[slotIdx].buffer + keySz * 2, curveId);
+    if (ret == 0) {
+        keySz = cacheMeta->len / 3;
+        ret = wc_ecc_import_unsigned(key, cacheBuf, cacheBuf + keySz,
+            cacheBuf + keySz * 2, curveId);
```

Comment:
while we're here, can we `const`ify the magic numbers with self describing names  just to make it explicitly clear what is going on at a glance

Reply:
this one can't be const because it depends on the length of the particular ecc key pulled from the metadata, will make a descriptive variable for it though